sgkit/stats/popgen.py

```diff
@@ -0,0 +1,166 @@
+from typing import Hashable
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from xarray import DataArray, Dataset
+
+from .aggregation import count_variant_alleles
+
+
+def diversity(
+    ds: Dataset, allele_counts: Hashable = "variant_allele_count",
+) -> DataArray:
+    """Compute diversity from allele counts.
+
+    Because we're not providing any arguments on windowing, etc,
+    we return the total over the whole region. Maybe this isn't
+    the behaviour we want, but it's a starting point. Note that
+    this is different to the tskit default behaviour where we
+    normalise by the size of windows so that results
+    in different windows are comparable. However, we don't have
+    any information about the overall length of the sequence here
+    so we can't normalise by it.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Genotype call dataset.
+    allele_counts : Hashable
+        allele counts to use or calculate.
+
+    Returns
+    -------
+    DataArray
+        diversity value.
+    """
+    if len(ds.samples) < 2:
+        return xr.DataArray(np.nan)
+    if allele_counts not in ds:
+        ds = count_variant_alleles(ds)
```

Comment:
I'm not sure we should modify the passed-in dataset. Instead we should return a dataset (like in #217) and add the allele counts to that dataset.